crates/router/src/core/payments/helpers.rs

```diff
@@ -2540,3 +2540,16 @@ pub async fn get_additional_payment_data(
         }
     }
 }
+
+pub fn validate_customer_access(
+    auth_flow: services::AuthFlow,
+    request: &api::PaymentsRequest,
+) -> Result<(), errors::ApiErrorResponse> {
+    if auth_flow == services::AuthFlow::Client && request.customer_id.is_some() {
```

Comment:
Here we are directly resisting they can't pass customer_id itself. Please verify any dependent services will break because of this. Ideal case they can pass id but it should be same.

Reply:
done